torch_xla/distributed/xla_dist.py

```diff
@@ -529,6 +572,10 @@ def run(self, cmd):
       nargs='+',
       type=str,
       help='The python command to launch training including model parameters.')
+  parser.add_argument(
+      '--tpuvm-mode',
```

Comment:
Could we infer this from the GCE VM metadata?

Like in TPUVM setup we have different GCE metadata set like this one: http://shortn/_PG51FEqQIT

We can use this to get the VM metadata:
https://github.com/pytorch/xla/blob/7f71a2c81b0323285663fe191b123247482583bd/torch_xla/distributed/cluster.py#L211-L216

Similarly we could use `instance-id` metadata so that users don't even need to pass in the `--tpu` flag if ssh'ed onto a TPUVM.

Reply:
+1 to auto-detect the tpuvm mode, it would be good for usability.

Reply:
will check